tests_tf/test_utils_tf.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+import unittest
+import tensorflow as tf
+
+from cleverhans import utils_tf
+
+
+def numpy_kl_with_logits(q_logits, p_logits):
+    def numpy_softmax(logits):
```

Comment:
This version of the softmax is numerically unstable. Should subtract off the max of logits before applying the exp.